doc/introduction/data.rst

```diff
@@ -0,0 +1,162 @@
+
+.. code-block:: python
+
+    >>> H2_dataset[0].molecule
+    <pennylane.qchem.molecule.Molecule object at 0x7f890b409280>
+    >>> H2_dataset[0].hf_state
+    [1 1 0 0 0 0 0 0]
+
+The loaded data items are fully compatible with PennyLane. We can therefore
+use them directly in PennyLane workflow as follows:
+
+.. code-block:: python
+
+    >>> dev = qml.device('default.qubit',wires=H2_dataset[0].hamiltonian.wires)
+    >>> @qml.qnode(dev)
+    ... def circuit():
+    ...     return qml.expval(H2_dataset[0].hamiltonian)
+    >>> print(circuit())
+    2.173913043478261
+
+Dataset Structure
+-----------------
+
+You can call the 
+:func:`~.pennylane.data.list_datasets` function to get a snapshot of the currently available data.
+This function returns a nested dictionary as we show below. Note that this example limits the results
+of the function calls for clarity and that as more data becomes available, the results of these
+function calls will change.
+
+.. code-block:: python
+
+    >>> available_data = qml.data.list_datasets()
+    >>> available_data.keys()
+    dict_keys(['qspin', 'qchem'])
+    >>> available_data['qchem']
+    dict_keys(['H2', 'HeH'])
+    >>> available_data['qchem']['H2']
+    dict_keys(['6-31G'])
+    >>> print(available_data['qchem']['H2']['6-31G'])
+    ['0.46', '1.16', '1.22', ...]
+
+Filtering Datasets
+------------------
+
+In the case that we only wish to download or load portions of a large dataset, we can specify the desired attributes.
+For example, we can download only the molecule and Hamiltonian of a dataset as follows:
+
+.. code-block:: python
+
+    >>> H2_hamiltonian = qml.data.load(data_type="qchem", molname="H2", basis="6-31G", bondlength="0.46", attributes=["molecule", "hamiltonian"])
+    >>> H2_hamiltonian
+    <Hamiltonian: terms=185, wires=[0, 1, 2, 3, 4, 5, 6, 7]>
+
+:html:`<div class="summary-table">`
+
+Creating Custom Datasets
+------------------------
+
+The functionality in :mod:`~.data` also includes creating and reading custom-made datasets.
+To create a dataset, we can do the following:
+
+.. code-block:: python
+
+    >>> example_hamiltonian = qml.Hamiltonian(coeffs=[1,0.5], observables=[qml.PauliZ(wires=0),qml.PauliX(wires=1)])
+    >>> example_energies, _ = np.linalg.eigh(qml.matrix(example_hamiltonian))
+    >>> example_dataset = qml.data.Dataset(dataset_name = 'An example dataset',hamiltonian=example_hamiltonian,energies=example_energies)
+    >>> example_dataset.dataset_name
+    'An example dataset'
+    >>> example_dataset.hamiltonian
+      (0.5) [X1]
+    + (1) [Z0]
+    >>> example_dataset.energies
+    array([-1.5, -0.5,  0.5,  1.5])
+
+We can then write this :class:`~.data.Dataset` to storage and read it as follows:
+
+.. code-block:: python
+    
+    >>> example_dataset.write('./path/to/dataset.dat')
+    >>> read_dataset = qml.data.Dataset()
+    >>> read_dataset.read('./path/to/dataset.dat')
```

Comment:
Yes, we should be able to read in one line. Looks weird otherwise.

Reply:
so.. should I add that `from_file` helper function after all?